### PR TITLE
[JBPM-9178] Immutable SpringBoot Deployment

### PR DESCRIPTION
**Pull request by elguardian**

### CODE REVIEW

**Comment by elguardian** [comment]
depends on https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1348
https://github.com/kiegroup/droolsjbpm-knowledge/pull/443
https://github.com/kiegroup/kie-soup/pull/142
https://github.com/kiegroup/drools/pull/2931

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieContainerInstanceImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+public class InmutableSpringBootKieContainerInstanceImpl extends KieContainerInstanceImpl {
+
+    public InmutableSpringBootKieContainerInstanceImpl(String containerId, KieContainerStatus status, InternalKieContainer kieContainer, ReleaseId releaseId, KieServerImpl kieServer) {
+        super(containerId, status, kieContainer, releaseId, kieServer);
+    }
+
+    @Override
+    protected boolean updateReleaseId() {
+        return false;
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieContainerInstanceImpl.java:34]: why false?

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    public class SpringBootKieServerController implements KieServerController {
+
+        @Override
+        public void disconnect(KieServerInfo serverInfo) {}
+
+        @Override
+        public KieServerSetup connect(KieServerInfo serverInfo) {
+            KieServerSetup serverSetup = new KieServerSetup();
+            serverSetup.setContainers(containers.stream().collect(Collectors.toSet()));
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java:69]: Lambda can be replaced with:

```suggestion
            serverSetup.setContainers(new HashSet<>(containers));
```

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    protected InternalKieContainer createInternalKieContainer(String containerId, ReleaseId releaseId, KieModuleMetaData metaData) {
+        return (InternalKieContainer) ks.newKieClasspathContainer(containerId, metaData.getClassLoader(), releaseId);
+    }
+
+    @Override
+    protected KieContainerInstanceImpl createContainerInstanceImpl(String containerId, ReleaseId releaseId) {
+        return new InmutableSpringBootKieContainerInstanceImpl(containerId, KieContainerStatus.CREATING, null, releaseId, this);
+    }
+
+    public class SpringBootKieServerController implements KieServerController {
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java:61]: Shouldn't we restrict access to this inner class? maybe `private` instead?

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        @Override
+        public void disconnect(KieServerInfo serverInfo) {}
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java:64]: Just add a small comment stating why this method body is empty.

**Comment by elguardian** [comment]
Jenkins retest this please

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/ArtifactItem.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
+package org.kie.maven.plugin;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/ArtifactItem.java:1]: done

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageMojo.java [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        DefaultArtifactCoordinate artifactCoordinate = new DefaultArtifactCoordinate();
+        artifactCoordinate.setGroupId(dependableCoordinate.getGroupId());
+        artifactCoordinate.setArtifactId(dependableCoordinate.getArtifactId());
+        artifactCoordinate.setVersion(dependableCoordinate.getVersion());
+        artifactCoordinate.setClassifier(dependableCoordinate.getClassifier());
+        artifactCoordinate.setExtension(artifactHandler.getExtension());
+        return artifactCoordinate;
+    }
+
+    ArtifactRepository parseRepository(String repo, ArtifactRepositoryPolicy policy) throws MojoFailureException {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageMojo.java:189]: changed

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageMojo.java [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+import org.apache.maven.shared.dependencies.resolve.DependencyResolver;
+import org.apache.maven.shared.dependencies.resolve.DependencyResolverException;
+import org.apache.maven.shared.utils.StringUtils;
+import org.codehaus.plexus.archiver.manager.ArchiverManager;
+
+@Mojo(name = "package-dependencies-kjar",
+      defaultPhase = LifecyclePhase.PREPARE_PACKAGE,
+      threadSafe = true,
+      requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class PackageMojo extends AbstractKieMojo {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageMojo.java:64]: PackageKjarDependenciesMojo sounds good.

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 8]
```diff
@@ -223,7 +224,7 @@ public void destroy() {
         return this.context.getServerExtensions();
     }
 
-    protected KieServerInfo getInfoInternal() {
+    public KieServerInfo getInfoInternal() {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:227]: removed a set as protected

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 9]
```diff
@@ -360,6 +362,14 @@ protected KieServerInfo getInfoInternal() {
         }
     }
 
+    protected KieContainerInstanceImpl createContainerInstanceImpl(String containerId, ReleaseId releaseId) {
+        return new KieContainerInstanceImpl(containerId, KieContainerStatus.CREATING, null, releaseId, this);
+    }
+
+    protected InternalKieContainer createInternalKieContainer(String containerId, ReleaseId releaseId, KieModuleMetaData metadata) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:369]: metadata is required for inmutable spring boot create Internal Kie Container.

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieContainerInstanceImpl.java (hunk 1 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieContainerInstanceImpl.java:34]: class path container does not allow updates.

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java [hunk 10]
```diff
@@ -420,9 +423,13 @@ public void createContainer(String id, KieContainerInstance kieContainerInstance
                 unit.setMergeMode(MergeMode.valueOf(mergeMode));
             }
             String ksession = config.getConfigItemValue(KieServerConstants.PCFG_KIE_SESSION);
-            unit.setKsessionName(ksession);
+            if (ksession != null) {
+                unit.setKsessionName(ksession);
+            }
             String kbase = config.getConfigItemValue(KieServerConstants.PCFG_KIE_BASE);
-            unit.setKbaseName(kbase);
+            if (kbase != null) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java:430]: ok

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java (hunk 2 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java:69]: ok

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java (hunk 4 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/InmutableSpringBootKieServerImpl.java:64]: ok

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 11]
```diff
@@ -974,8 +989,8 @@ private Message updateKieContainerToVersion(KieContainerInstanceImpl kci, Releas
         return response;
     }
 
-    private void prepareUpdateExtensions(KieContainerInstanceImpl kci, ReleaseId releaseId, List<Message> messages, boolean resetBeforeUpdate) {
-        Map<String, Object> parameters = getReleaseUpdateParameters(releaseId, messages, resetBeforeUpdate);
+    private void prepareUpdateExtensions(KieModuleMetaData metadata, KieContainerInstanceImpl kci, ReleaseId releaseId, List<Message> messages, boolean resetBeforeUpdate) {
+        Map<String, Object> parameters = getReleaseUpdateParameters(null, releaseId, kci.getKieContainer(), messages, resetBeforeUpdate);
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:993]: any update here?

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 12]
```diff
@@ -1029,7 +1052,7 @@ private void updateExtensions(KieContainerInstanceImpl kci, ReleaseId releaseId,
         return parameters;
     }
 
-    protected KieServerController getController() {
+    public KieServerController getController() {
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:1055]: any update here?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java [hunk 13]
```diff
@@ -420,9 +423,13 @@ public void createContainer(String id, KieContainerInstance kieContainerInstance
                 unit.setMergeMode(MergeMode.valueOf(mergeMode));
             }
             String ksession = config.getConfigItemValue(KieServerConstants.PCFG_KIE_SESSION);
-            unit.setKsessionName(ksession);
+            if (ksession != null) {
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java:426]: any update here?

File: kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieServerExtension.java [hunk 14]
```diff
@@ -89,12 +89,16 @@ public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
...
+
         // do any other bootstrapping rule service requires
         Set<Class<?>> extraClasses = new HashSet<Class<?>>();
 
         // create kbases so declared types can be created
         Collection<String> kbases = kieContainerInstance.getKieContainer().getKieBaseNames();
         for (String kbase : kbases) {
+            if (kbase.startsWith("KiePMML")) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieServerExtension.java:101]: any update on this?

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/ContainerManager.java [hunk 15]
```diff
@@ -34,11 +35,14 @@ public void installContainers(KieServerImpl kieServer, Set<KieContainerResource>
     }
 
     public void installContainersSync(KieServerImpl kieServer, Set<KieContainerResource> containers, KieServerState currentState, KieServerSetup kieServerSetup) {
-        logger.info("About to install containers '{}' on kie server '{}'", containers, kieServer);
         if (containers == null) {
             kieServer.markAsReady();
             return;
         }
+
+        String str = containers.stream().map(e -> e.toString()).collect(Collectors.joining("\n\t"));
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/ContainerManager.java:43]: any update on this?

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/ImmutableSpringBootKieServerImpl.java [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+    protected InternalKieContainer createInternalKieContainer(String containerId, ReleaseId releaseId, KieModuleMetaData metaData) {
+        return (InternalKieContainer) ks.newKieClasspathContainer(containerId, metaData.getClassLoader(), releaseId);
+    }
+
+    @Override
+    protected KieContainerInstanceImpl createContainerInstanceImpl(String containerId, ReleaseId releaseId) {
+        return new InmutableSpringBootKieContainerInstanceImpl(containerId, KieContainerStatus.CREATING, null, releaseId, this);
+    }
+
+    public class SpringBootKieServerController implements KieServerController {
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/ImmutableSpringBootKieServerImpl.java:61]: Shouldn't we restrict access modifier? it's only used internally, so maybe it's better to declare as `private`?

**Comment by elguardian** [comment]
Jenkins retest this please.

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+
+            if (remoteRepositories != null) {
+                // Use the same format as in the deploy plugin id::layout::url
+                String[] repos = StringUtils.split(remoteRepositories, ",");
+                for (String repo : repos) {
+                    repoList.add(parseRepository(repo, always));
+                }
+            }
+
+            File outputFolder = new File(outputDirectory + "/KIE-INF/lib");
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java:139]: I will check if it is possible to workaround spring boot plugin. don't commit to anything

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+
+@Mojo(name = "package-dependencies-kjar",
+      defaultPhase = LifecyclePhase.PREPARE_PACKAGE,
+      threadSafe = true,
+      requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class PackageKjarDependenciesMojo extends AbstractKieMojo {
+
+    private static final Pattern ALT_REPO_SYNTAX_PATTERN = Pattern.compile("(.+)::(.*)::(.+)");
+
+    @Parameter(defaultValue = "${project.build.outputDirectory}", required = true)
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java:68]: sure

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java (hunk 17 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java:139]: nope

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+    private String outputDirectory;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    private MavenProject project;
+
+    @Parameter
+    private String classifier;
+
+    @Component
+    private MavenProjectHelper projectHelper;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java:78]: removed

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java [hunk 20]
```diff
@@ -0,0 +1,223 @@
...
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    private MavenSession session;
+
+    @Component(role = ArchiverManager.class)
+    private ArchiverManager archiverManager;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java:84]: removed

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java [hunk 21]
```diff
@@ -0,0 +1,223 @@
...
+
+    @Parameter
+    private MavenArchiveConfiguration archive = new MavenArchiveConfiguration();
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-maven-plugin/src/main/java/org/kie/maven/plugin/PackageKjarDependenciesMojo.java:87]: removed

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/ImmutableSpringBootKieServerImpl.java (hunk 16 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/ImmutableSpringBootKieServerImpl.java:61]: sure

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/ContainerManager.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/ContainerManager.java:43]: ok

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/InmutableContainerStartupStrategy.java [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+        KieServerSetup kieServerSetup = kieServer.getController().connect(kieServerInfo);
+
+        Set<KieContainerResource> containers = kieServerSetup.getContainers();
+
+        kieServer.addServerStatusMessage(kieServerInfo);
+        if (Boolean.parseBoolean(currentState.getConfiguration().getConfigItemValue(KieServerConstants.CFG_SYNC_DEPLOYMENT, "false"))) {
+            containerManager.installContainersSync(kieServer, containers, currentState, kieServerSetup);
+        } else {
+            containerManager.installContainers(kieServer, containers, currentState, kieServerSetup);
+        }
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/InmutableContainerStartupStrategy.java:44]: mimic the rest of the startup strategy containers.

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:993]: changed

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java:426]: that should not be possible.

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by elguardian** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
@elguardian - CI is failing due to some duplicated classes in `kie-takari-plugin` module. https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-takari-plugin/pom.xml#L39-L42

Exclusions are needed, basically almost the same as the ones already defined in this pom.xml

https://github.com/kiegroup/droolsjbpm-integration/pull/2119/files#diff-380451c08e393170be9a2d53b0d1c369R38-R51

See error stack trace below:

[2020-08-05T06:52:54.945Z] [WARNING] Rule 0: org.apache.maven.plugins.enforcer.BanDuplicateClasses failed with message:
[2020-08-05T06:52:54.945Z] Duplicate classes found:
[2020-08-05T06:52:54.945Z] 
[2020-08-05T06:52:54.945Z]   Found in:
[2020-08-05T06:52:54.945Z]     javax.inject:javax.inject:jar:1:compile
[2020-08-05T06:52:54.945Z]     org.sonatype.sisu:sisu-inject-bean:jar:2.3.0:compile
[2020-08-05T06:52:54.945Z]   Duplicate classes:
[2020-08-05T06:52:54.945Z]     javax/inject/Inject.class
[2020-08-05T06:52:54.945Z] 
[2020-08-05T06:52:54.945Z]   Found in:
[2020-08-05T06:52:54.945Z]     aopalliance:aopalliance:jar:1.0:compile
[2020-08-05T06:52:54.945Z]     org.sonatype.sisu:sisu-inject-bean:jar:2.3.0:compile
[2020-08-05T06:52:54.945Z]   Duplicate classes:
[2020-08-05T06:52:54.945Z]     org/aopalliance/aop/AspectException.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/ConstructorInterceptor.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/MethodInvocation.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/MethodInterceptor.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/Interceptor.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/Joinpoint.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/Invocation.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/aop/Advice.class
[2020-08-05T06:52:54.945Z]     org/aopalliance/intercept/ConstructorInvocation.class

**Comment by afalhambra** [comment]
Failing with similar issue at `kie-plugin-testing` module now:

[2020-08-06T15:05:58.901Z] Duplicate classes found:
[2020-08-06T15:05:58.901Z] 
[2020-08-06T15:05:58.901Z]   Found in:
[2020-08-06T15:05:58.901Z]     javax.inject:javax.inject:jar:1:compile
[2020-08-06T15:05:58.901Z]     org.sonatype.sisu:sisu-inject-bean:jar:2.3.0:compile
[2020-08-06T15:05:58.901Z]   Duplicate classes:
[2020-08-06T15:05:58.901Z]     javax/inject/Inject.class
[2020-08-06T15:05:58.901Z] 
[2020-08-06T15:05:58.901Z]   Found in:
[2020-08-06T15:05:58.901Z]     aopalliance:aopalliance:jar:1.0:compile
[2020-08-06T15:05:58.901Z]     org.sonatype.sisu:sisu-inject-bean:jar:2.3.0:compile
[2020-08-06T15:05:58.901Z]   Duplicate classes:
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/ConstructorInterceptor.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/MethodInvocation.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/MethodInterceptor.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/Invocation.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/aop/AspectException.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/Interceptor.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/Joinpoint.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/aop/Advice.class
[2020-08-06T15:05:58.901Z]     org/aopalliance/intercept/ConstructorInvocation.class
[2020-08-06T15:05:58.901Z] 
[2020-08-06T15:05:58.901Z]   Found in:
[2020-08-06T15:05:58.901Z]     org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.2:compile
[2020-08-06T15:05:58.901Z]     org.sonatype.sisu:sisu-inject-bean:jar:2.3.0:compile
[2020-08-06T15:05:58.901Z]   Duplicate classes:
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/Description.class
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/Nullable.class
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/Mediator.class
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/BeanScanning.class
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/BeanEntry.class
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/EagerSingleton.class
[2020-08-06T15:05:58.901Z]     org/sonatype/inject/Parameters.class

**Pull request merged**